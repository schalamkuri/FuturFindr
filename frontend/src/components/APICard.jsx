import {Card, Button} from 'react-bootstrap';

const APICard = (inputAPI) => {
  return (
    <Card>
      <Card.Img variant="top" src={inputAPI?.inputAPI?.pic} />
            <Card.Body>
                <Card.Title>{inputAPI.inputAPI?.name}<br /></Card.Title>
                <Card.Text>{inputAPI.inputAPI?.desc}</Card.Text>
                <Button variant="primary" href={inputAPI.inputAPI?.link}>Link</Button>
            </Card.Body>
    </Card>
  );
}

export default APICard;