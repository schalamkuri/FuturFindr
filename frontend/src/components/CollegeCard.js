import {
    Grid,
    Card,
    Typography,
    CardMedia,
} from "@mui/material";


const CollegeCard = (props) => {

    let name = props.name
    let tuition = props.tuition
    let media = props.media


    return (
        <Grid item xs={3}>
            <Card className='college-card' color="purple">
                <CardMedia
                sx={{ height: 140 }}
                image={media}>
                </CardMedia>
               <Typography>{name}</Typography>
               <Typography>Tuition: ${tuition}</Typography>
            </Card>
        </Grid >

    );
}


export default CollegeCard;