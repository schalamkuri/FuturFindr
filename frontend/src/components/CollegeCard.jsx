import React from 'react'
import Card from 'react-bootstrap/Card';

function CollegeCard(props) {
	return (
		<Card className="card border-dark mb-3" style={{height: "90%"}}>
			<Card.Img variant="top" src={props.img_url ? props.img_url : "https://www.convergemedia.org/wp-content/uploads/2017/01/academia-1000.png"}/>
			<Card.Body>
				<Card.Title>{props.name}</Card.Title>
				<Card.Text>
					Instate Tuition: ${props.instateTuition ? props.instateTuition : "~"} <br></br>
					Outstate Tuition: ${props.outstateTuition ? props.outstateTuition : "~"} <br></br>
					Admission Rate: %{props.admissionRate ? Math.trunc(props.admissionRate * 100) : "~"} <br></br>
					City: {props.city} <br></br>
				</Card.Text>
				<a href={`/colleges/${props.id}`} class="stretched-link"></a>
			</Card.Body>
		</Card>
	);
}

export default CollegeCard;