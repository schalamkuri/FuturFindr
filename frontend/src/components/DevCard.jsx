import Card from 'react-bootstrap/Card';
import Tab from 'react-bootstrap/Tab';
import Tabs from 'react-bootstrap/Tabs';

const DevCard = (inputUser) => {
  return (
    <Card>
      <Card.Img variant="top" src={inputUser?.inputUser?.pic ?? "fsf"} />
        <Tabs
        defaultActiveKey="about"
        id="devcard-tabs"
        className="mb-3"
        justify
        >
            <Tab eventKey="about" title="About">
                <Card.Body>
                        <Card.Title>{inputUser?.inputUser?.name}</Card.Title>
                        <Card.Text>Role: {inputUser?.inputUser?.role}</Card.Text>
                        <Card.Text>{inputUser?.inputUser?.bio}</Card.Text>
                </Card.Body>
            </Tab>
            <Tab eventKey="GitLab" title="GitLab"> 
                <Card.Body>
                    <Card.Title>@{inputUser?.inputUser?.gitlab}<br /></Card.Title>
                    Commits: {inputUser?.inputUser?.commits}<br />
                    Issues: {inputUser?.inputUser?.issues}<br />
                    Unit Tests: {inputUser?.inputUser?.tests}
                </Card.Body>
            </Tab>
        </Tabs>
    </Card>
  );
}

export default DevCard;