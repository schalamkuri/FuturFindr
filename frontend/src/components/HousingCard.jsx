import React from 'react'
import Card from 'react-bootstrap/Card';

function HousingCard(props) {
  return (
    <Card className="card border-dark mb-3" style={{height: "90%"}}>
      <Card.Img variant="top" src={props.image} />
      <Card.Body>
        <Card.Title>{props.address}</Card.Title>
        <Card.Text>
            {props.type} <br></br>
            City: {props.city ? props.city : "~"} <br></br>
            Monthly Rent: ${props.price ? props.price : "~"} <br></br>
            Beds: {props.beds ? props.beds : "~"} <br></br>
            Baths {props.baths ? props.baths : "~"} <br></br>
        </Card.Text>
        <a href={`/housing/${props.id}`} className="stretched-link"></a>
      </Card.Body>
    </Card>
  )
}

export default HousingCard
