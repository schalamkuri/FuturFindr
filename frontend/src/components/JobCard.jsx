import React from "react";
import Card from "react-bootstrap/Card";

function JobCard(props) {
  return (
    <Card className="card border-dark mb-3" style={{ height: "90%" }}>
      <Card.Img variant="top" src={props.image} />
      <Card.Body>
        <Card.Title>{props.title}</Card.Title>
        <Card.Text>
          Company: {props.company} <br></br>
          Industry: {props.category} <br></br>
          Location: {props.location} <br></br>
          Pay range: ${props.salaryMin}-{props.salaryMax} <br></br>
          Date Posted: {props?.datePosted?.split("T")[0]} <br></br>
        </Card.Text>
        <a href={`/jobs/${props.id}`} className="stretched-link"></a>
      </Card.Body>
    </Card>
  );
}

export default JobCard;
