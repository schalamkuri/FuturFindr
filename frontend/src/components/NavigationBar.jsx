import Logo from '../Assets/fortune-teller.png';
import { Nav, Navbar, Button, Form, Container } from "react-bootstrap";

// Code is based off of geojobs navigation bar
// https://gitlab.com/sarthaksirotiya/cs373-idb/-/blob/main/front-end/src/components/GlobalNavbar/GlobalNavbar.jsx
  
  const NavigationBar = () => {

    const handleSubmit = (event) => {
      event.preventDefault()
      window.location.assign(`/search/${event.currentTarget.query.value}`)
    }
    return (
      <Navbar
      
        variant="light"
        expand="lg"
        style={{
          backgroundColor: "#84bcdc",
          display: "flex",
          alignItems: "right",
        }}
      >
        <Container>
          <Navbar.Brand href="/">
            <div style={{ display: "flex", justifyContent: "space-evenly" }}>
              <img src={Logo} height={"50px"} width={"50px"} className="img-fluid"></img>
              <h2>FuturFindr</h2>
            </div>
          </Navbar.Brand>
          <Navbar.Toggle aria-controls="basic-navbar-nav" />
          <Navbar.Collapse style={{ alignItems: "left" }}>
            <Nav className="mr-auto">
              <Nav.Link href="/about">
                About
              </Nav.Link>
              <Nav.Link href="/colleges">
                Colleges
              </Nav.Link>
              <Nav.Link href="/jobs">
                Jobs
              </Nav.Link>
              <Nav.Link href="/housing">
                Housing
              </Nav.Link>
            </Nav>
            <Container className="d-flex justify-content-end">
            <Form onSubmit={handleSubmit} className="d-flex">
              <Form.Control
                style={{ width: "20vw" }}
                type="search"
                name="query"
                placeholder="Colleges, Jobs, & Housing"
                className="me-2"
                aria-label="Search"
              />
              <Button type="submit" variant="outline-secondary">Search</Button>
              {/*style={{color: '#3d405b'}}*/}
            </Form>
          </Container>


          </Navbar.Collapse>
        </Container>
      </Navbar>
    );
  };
  
  export default NavigationBar;
  