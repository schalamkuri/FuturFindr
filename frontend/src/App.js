import NavigationBar from "./components/NavigationBar";
import RouteSwitch from "./RouteSwitch";

function App() {
  return (
    <div className="App">
      <NavigationBar />
      <RouteSwitch />
    </div>
  );
}

export default App;
