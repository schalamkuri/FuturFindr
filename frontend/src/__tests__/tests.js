import renderer from "react-test-renderer";

import JobCard from "../components/JobCard";
import HousingCard from "../components/HousingCard";
import CollegeCard from "../components/CollegeCard";
import DevCard from "../components/DevCard";
import ToolCard from "../components/ToolCard";
import APICard from "../components/APICard"
import NavigationBar from "../components/NavigationBar";
import About from "../Views/about";

import CollegeInstance from "../Views/CollegeInstance";
import HousingInstance from "../Views/HousingInstance";
import RouteSwitch from "../RouteSwitch";
import App from "../App";

// import placeholder from "../assets/placeholder/avatar.png";
// import App from "../App";
// import RouteSwitch from "../RouteSwitch";

it("CollegeCard Initial correctly", () => {
  const college = {
    name : "name",
    instateTuition : "instateTuition",
    outstateTuition : "outstateTuition",
    admissionRate: "admissionRate",
    city: "city",
    img_url: "https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.utexas.edu%2Fabout%2Foverview&psig=AOvVaw285B_AsAjQqCavtg_9SRdT&ust=1678332012643000&source=images&cd=vfe&ved=0CA8QjRxqFwoTCLjVp42wy_0CFQAAAAAdAAAAABAE",
  };
  const component = renderer.create(<CollegeCard college={college} />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("HousingCard initializes correctly", () => {
  const housing = {
    address: "address",
    type: "type",
    city: "city",
    price: 1000,
    beds: 5,
    baths: 2,
  };
  const component = renderer.create(<HousingCard housing={housing} />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});



it("JobCard initializes correctly", () => {
  const job = {
    title: "title",
    company: "company",
    category: "category",
    location: 'location',
    id: "id"
  };
  const component = renderer.create(<JobCard job={job} />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("DevCard initializes correctly", () => {
  const dev = {
    pic: "https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.utexas.edu%2Fabout%2Foverview&psig=AOvVaw285B_AsAjQqCavtg_9SRdT&ust=1678332012643000&source=images&cd=vfe&ved=0CA8QjRxqFwoTCLjVp42wy_0CFQAAAAAdAAAAABAE",
    name: "name",
    role: "role",
    bio: "bio",
    gitlab: "gitlab",
    commits: "commits"
  };
  const component = renderer.create(<DevCard dev={dev} />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("ToolCard initializes correctly", () => {
  const tool = {
    name: "name",
    role: "role",
    bio: "bio",
    gitlab: "gitlab",
    commits: "commits"
  };
  const component = renderer.create(<ToolCard tool={tool} />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("APICard initializes correctly", () => {
  const api = {
    name: "name",
    role: "role",
    bio: "bio",
    gitlab: "gitlab",
    commits: "commits"
  };
  const component = renderer.create(<APICard api={api} />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("NavigationBar initializes correctly", () => {
  const component = renderer.create(<NavigationBar />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("About initializes correctly", () => {
  const component = renderer.create(<About />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

// it("CollegeInstance Initial correctly", () => {
//   const college = {
//     name : "name",
//     instateTuition : "instateTuition",
//     outstateTuition : "outstateTuition",
//     admissionRate: "admissionRate",
//     city: "city",
//     img_url: "https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.utexas.edu%2Fabout%2Foverview&psig=AOvVaw285B_AsAjQqCavtg_9SRdT&ust=1678332012643000&source=images&cd=vfe&ved=0CA8QjRxqFwoTCLjVp42wy_0CFQAAAAAdAAAAABAE",
//   };
//   const component = renderer.create(<CollegeInstance college={college} />);

//   let tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });

it("RouteSwitch initializes correctly", () => {
  const component = renderer.create(<App />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


it("RouteSwitch initializes correctly", () => {
  const component = renderer.create(<RouteSwitch />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

// it("HousingCard initializes correctly", () => {
//   const housing = {
//     address: "address",
//     type: "type",
//     city: "city",
//     price: 1000,
//     beds: 5,
//     baths: 2,
//   };
//   const component = renderer.create(<HousingInstance housing={housing} />);

//   let tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });

