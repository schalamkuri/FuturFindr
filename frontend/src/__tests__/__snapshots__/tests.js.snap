// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APICard initializes correctly 1`] = `
<div
  className="card"
>
  <img
    className="card-img-top"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      <br />
    </div>
    <p
      className="card-text"
    />
    <button
      className="btn btn-primary"
      disabled={false}
      type="button"
    >
      Link
    </button>
  </div>
</div>
`;

exports[`About initializes correctly 1`] = `
<div
  className="vstack"
>
  <div
    className="p-4 container"
  >
    <h2
      className="d-flex justify-content-left"
    >
      What is FuturFindr?
    </h2>
    <p>
      FuturFindr (pronounced future finder) is a platform that assists users in planning out their future by helping them find the right place to advance their education, find the right place to work in the real world, or finding a place to live for either of those. This is an all in one platform that is applicable to anyone who is interested in advancing themselves, whether it be a high school student, or someone who is looking to potentially switch careers, or someone who just wants to find a place to live from where their place of work or school is at.
    </p>
    <p>
      FuturFindr uses a React front-end combined with an SQL database to scrape and combine API data and present it back to you as nice, clean information cards. By connecting careers, education, and living spaces, with cards filled with relevant information and links to the necessary places, this results with a service that can make advancing yourself easier (and save you from opening so many chrome tabs)!
    </p>
  </div>
  <div
    className="p-4 container"
  >
    <h2
      className="d-flex justify-content-left"
    >
      Developers
    </h2>
    <div
      className="row"
    >
      <div
        className="d-flex justify-content-center col"
      >
        Loading
      </div>
    </div>
  </div>
  <div
    className="p-4 e container"
  >
    <h3
      className="d-flex justify-content-center"
    >
      Gitlab Repository Stats
    </h3>
    <div
      className="row"
    >
      <div
        className="d-flex justify-content-center col"
      >
        Total Issues:
      </div>
      <div
        className="d-flex justify-content-center col"
      >
        Total Commits:
      </div>
      <div
        className="d-flex justify-content-center col"
      >
        Total Unit Tests:
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="d-flex justify-content-center col"
      >
        0
      </div>
      <div
        className="d-flex justify-content-center col"
      >
        0
      </div>
      <div
        className="d-flex justify-content-center col"
      >
        0
      </div>
    </div>
  </div>
  <div
    className="p-4 container"
  >
    <a
      href="https://documenter.getpostman.com/view/25807396/2s93CExcaC"
      target="_blank"
    >
      <h3
        className="d-flex justify-content-center"
      >
        Postman API Documentation
      </h3>
    </a>
  </div>
  <div
    className="p-4 container"
  >
    <h3
      className="d-flex justify-content-center"
    >
      Tools We Used:
    </h3>
    <div
      className="row"
    >
      <div
        className="d-flex justify-content-center col"
      >
        Loading
      </div>
    </div>
  </div>
  <div
    className="p-4 container"
  >
    <h3
      className="d-flex justify-content-center"
    >
      APIs We Scraped:
    </h3>
    <div
      className="row"
    >
      <div
        className="d-flex justify-content-center col"
      >
        Loading
      </div>
    </div>
  </div>
</div>
`;

exports[`CollegeCard Initial correctly 1`] = `
<div
  className="card border-dark mb-3 card"
  style={
    Object {
      "height": "90%",
    }
  }
>
  <img
    className="card-img-top"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    />
    <p
      className="card-text"
    >
      Instate Tuition: $
      ~
       
      <br />
      Outstate Tuition: $
      ~
       
      <br />
      Admission Rate: %
      ~
       
      <br />
      City: 
       
      <br />
    </p>
    <a
      class="stretched-link"
      href="/colleges/undefined"
    />
  </div>
</div>
`;

exports[`DevCard initializes correctly 1`] = `
<div
  className="card"
>
  <img
    className="card-img-top"
    src="fsf"
  />
  <ul
    className="mb-3 nav nav-tabs nav-justified"
    onKeyDown={[Function]}
    role="tablist"
  >
    <li
      className="nav-item"
      role="presentation"
    >
      <button
        aria-controls="devcard-tabs-tabpane-about"
        aria-selected={true}
        className="nav-link active"
        data-rr-ui-event-key="about"
        disabled={false}
        id="devcard-tabs-tab-about"
        onClick={[Function]}
        role="tab"
        type="button"
      >
        About
      </button>
    </li>
    <li
      className="nav-item"
      role="presentation"
    >
      <button
        aria-controls="devcard-tabs-tabpane-GitLab"
        aria-selected={false}
        className="nav-link"
        data-rr-ui-event-key="GitLab"
        disabled={false}
        id="devcard-tabs-tab-GitLab"
        onClick={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        GitLab
      </button>
    </li>
  </ul>
  <div
    className="tab-content"
  >
    <div
      aria-labelledby="devcard-tabs-tab-about"
      className="fade tab-pane active show"
      id="devcard-tabs-tabpane-about"
      role="tabpanel"
    >
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        />
        <p
          className="card-text"
        >
          Role: 
        </p>
        <p
          className="card-text"
        />
      </div>
    </div>
    <div
      aria-labelledby="devcard-tabs-tab-GitLab"
      className="fade tab-pane"
      id="devcard-tabs-tabpane-GitLab"
      role="tabpanel"
    >
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          @
          <br />
        </div>
        Commits: 
        <br />
        Issues: 
        <br />
        Unit Tests: 
      </div>
    </div>
  </div>
</div>
`;

exports[`HousingCard initializes correctly 1`] = `
<div
  className="card border-dark mb-3 card"
  style={
    Object {
      "height": "90%",
    }
  }
>
  <img
    className="card-img-top"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    />
    <p
      className="card-text"
    >
       
      <br />
      City: 
      ~
       
      <br />
      Monthly Rent: $
      ~
       
      <br />
      Beds: 
      ~
       
      <br />
      Baths 
      ~
       
      <br />
    </p>
    <a
      className="stretched-link"
      href="/housing/undefined"
    />
  </div>
</div>
`;

exports[`JobCard initializes correctly 1`] = `
<div
  className="card border-dark mb-3 card"
  style={
    Object {
      "height": "90%",
    }
  }
>
  <img
    className="card-img-top"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    />
    <p
      className="card-text"
    >
      Company: 
       
      <br />
      Industry: 
       
      <br />
      Location: 
       
      <br />
      Pay range: $
      -
       
      <br />
      Date Posted: 
       
      <br />
    </p>
    <a
      className="stretched-link"
      href="/jobs/undefined"
    />
  </div>
</div>
`;

exports[`NavigationBar initializes correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light"
  style={
    Object {
      "alignItems": "right",
      "backgroundColor": "#84bcdc",
      "display": "flex",
    }
  }
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <div
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-evenly",
          }
        }
      >
        <img
          className="img-fluid"
          height="50px"
          src="fortune-teller.png"
          width="50px"
        />
        <h2>
          FuturFindr
        </h2>
      </div>
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      style={
        Object {
          "alignItems": "left",
        }
      }
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rr-ui-event-key="/about"
          disabled={false}
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/colleges"
          disabled={false}
          href="/colleges"
          onClick={[Function]}
        >
          Colleges
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/jobs"
          disabled={false}
          href="/jobs"
          onClick={[Function]}
        >
          Jobs
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/housing"
          disabled={false}
          href="/housing"
          onClick={[Function]}
        >
          Housing
        </a>
      </div>
    </div>
  </div>
</nav>
`;

exports[`RouteSwitch initializes correctly 1`] = `
<div
  className="App"
>
  <link
    crossOrigin="anonymous"
    href="https://cdn.jsdelivr.net/npm/bootstrap@4.1.3/dist/css/bootstrap.min.css"
    integrity="sha384-MCw98/SFnGE8fJT3GXwEOngsV7Zt27NXFoaoApmYm81iuXoPkFOJwJ8ERdknLPMO"
    rel="stylesheet"
  />
  <nav
    className="navbar navbar-expand-lg navbar-light"
    style={
      Object {
        "alignItems": "right",
        "backgroundColor": "#84bcdc",
        "display": "flex",
      }
    }
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        <div
          style={
            Object {
              "display": "flex",
              "justifyContent": "space-evenly",
            }
          }
        >
          <img
            className="img-fluid"
            height="50px"
            src="fortune-teller.png"
            width="50px"
          />
          <h2>
            FuturFindr
          </h2>
        </div>
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        style={
          Object {
            "alignItems": "left",
          }
        }
      >
        <div
          className="mr-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-link"
            data-rr-ui-event-key="/about"
            disabled={false}
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key="/colleges"
            disabled={false}
            href="/colleges"
            onClick={[Function]}
          >
            Colleges
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key="/jobs"
            disabled={false}
            href="/jobs"
            onClick={[Function]}
          >
            Jobs
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key="/housing"
            disabled={false}
            href="/housing"
            onClick={[Function]}
          >
            Housing
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div>
    <div
      className="card"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      <img
        className="card-img"
        src="Splash.png"
      />
      <div
        className="bigText card-img-overlay"
      >
        <h1>
          What's in your future?
        </h1>
      </div>
    </div>
    <div
      className="card-group"
    >
      <div
        className="card border-dark mb-3 card"
      >
        <img
          className="card-img-top"
          src="Campus.jpg"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Colleges
          </div>
          <p
            className="card-text"
          >
            Find the perfect place to further your education!
          </p>
        </div>
        <a
          className="stretched-link"
          href="colleges"
        />
      </div>
      <div
        className="card border-dark mb-3 card"
      >
        <img
          className="card-img-top"
          src="Jobs.jpg"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Jobs
          </div>
          <p
            className="card-text"
          >
            Match with the place where your skills are needed!
          </p>
        </div>
        <a
          className="stretched-link"
          href="jobs"
        />
      </div>
      <div
        className="card border-dark mb-3 card"
      >
        <img
          className="card-img-top"
          src="Apt.jpg"
        />
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            Housing
          </div>
          <p
            className="card-text"
          >
            The home of your dreams is waiting!
          </p>
        </div>
        <a
          className="stretched-link"
          href="housing"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`RouteSwitch initializes correctly 2`] = `
<div>
  <div
    className="card"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <img
      className="card-img"
      src="Splash.png"
    />
    <div
      className="bigText card-img-overlay"
    >
      <h1>
        What's in your future?
      </h1>
    </div>
  </div>
  <div
    className="card-group"
  >
    <div
      className="card border-dark mb-3 card"
    >
      <img
        className="card-img-top"
        src="Campus.jpg"
      />
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          Colleges
        </div>
        <p
          className="card-text"
        >
          Find the perfect place to further your education!
        </p>
      </div>
      <a
        className="stretched-link"
        href="colleges"
      />
    </div>
    <div
      className="card border-dark mb-3 card"
    >
      <img
        className="card-img-top"
        src="Jobs.jpg"
      />
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          Jobs
        </div>
        <p
          className="card-text"
        >
          Match with the place where your skills are needed!
        </p>
      </div>
      <a
        className="stretched-link"
        href="jobs"
      />
    </div>
    <div
      className="card border-dark mb-3 card"
    >
      <img
        className="card-img-top"
        src="Apt.jpg"
      />
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          Housing
        </div>
        <p
          className="card-text"
        >
          The home of your dreams is waiting!
        </p>
      </div>
      <a
        className="stretched-link"
        href="housing"
      />
    </div>
  </div>
</div>
`;

exports[`ToolCard initializes correctly 1`] = `
<div
  className="card"
>
  <img
    className="card-img-top"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      <br />
    </div>
    <p
      className="card-text"
    />
    <a
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
      target="_blank"
    >
      Link
    </a>
  </div>
</div>
`;
